pkg/services/query/query.go

```diff
@@ -82,11 +82,13 @@ func (s *Service) QueryData(ctx context.Context, user *user.SignedInUser, skipCa
 func (s *Service) QueryDataMultipleSources(ctx context.Context, user *user.SignedInUser, skipCache bool, reqDTO dtos.MetricRequest, handleExpressions bool) (*backend.QueryDataResponse, error) {
```

Comment:
why do we have two functions for this?  can we just merge them into QueryData?

Reply:
I think my only hesitation with that is `QueryData()` is used elsewhere with the expectation that it does not support mixed ds queries. Maybe its minor, but I wasn't sure if I should be break that expectation. By having the separate `QueryMultipleSource` there was the peace of mind that it was only being used by public dashboards, at least for now. Thoughts?

Reply:
I would not think of it as "breaking expectations" so much as supporting a new feature :)    

It was not supported before because it was not needed and do do it efficiently requires a bit more effort (even in the path you have here)

Reply:
Since the way we're handling multiple ds queries here isn't the most efficient yet, maybe we should leave this as-is and refactor it separately in a different card? Reasoning for that being:
- This solves the problem with expressions for public dashboards for the time being and lets us get it out the door quicker (this pr should probably be in a patch release actually).
- I'm happy to refactor it the efficient way, but I'm new to Go and it's concurrency model, so this refactor will probably take a little longer than usual. So may as well split that out.

Reply:
@owensmallwood happy to jump in and work with you on the concurrency model for this. I don't think we're in a particular rush to get this out the door so I think it would be efficient to verify things are working with a test harness and circle back on concurrency.